### PR TITLE
Fix plotting of values containers in an array

### PR DESCRIPTION
**Pull request by thomas-moulard**
Before this commit, attempting to plot a value in an array would trigger the following exception:

```sh
Traceback (most recent call last):
  File "/opt/ros/eloquent/lib/python3.6/site-packages/rqt_plot/plot_widget.py", line 259, in on_topic_edit_textChanged
    plottable, message = is_plottable(self._node, topic_name)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/rqt_plot/plot_widget.py", line 154, in is_plottable
    fields, message = get_plot_fields(node, topic_name)
  File "/opt/ros/eloquent/lib/python3.6/site-packages/rqt_plot/plot_widget.py", line 137, in get_plot_fields
    for slot, slot_type in field_class.get_fields_and_field_types().items():
AttributeError: 'NoneType' object has no attribute 'get_fields_and_field_types'
```
    
Line 129 in `plot_widget.py`:

```py
    field_class = message_helpers.get_message_class(slot_type)
```

...would be executed with slot_type set to `sequence<foo_msgs/Foo>` which
returns a `None` value. field_class is set to `None` and the call to
`get_fields_and_field_types` is then failing.

This pull request adds an extra layer of logic before calling get_message_class
to ensure that, if the type is an array, we extract the inner array type
instead of attempting to run get_message_class on the DDS sequence type
directly.

### CODE REVIEW

File: src/rqt_plot/plot_widget.py [hunk 1]
```diff
@@ -114,6 +115,14 @@ def get_plot_fields(node, topic_name):
         slot_type, slot_is_array, array_size = _parse_type(slot_type)
         is_array = slot_is_array and field_index is None
 
+        # If the field is an array, we need to update slot_type to extract
+        # the inner type, e.g. sequence<foo_msgs/Foo> -> foo_msgs/Foo
+        # See https://design.ros2.org/articles/mapping_dds_types.html
+        if is_array:
+            sequence_inner_type_regexp = re.match('sequence<(.*)>', slot_type)
+            if sequence_inner_type_regexp:
+                slot_type = sequence_innter_type_regexp.groupdict()['slot_type']
```
Comment (dirk-thomas) [inline src/rqt_plot/plot_widget.py]: This looks like a spelling mistake. Please update and afterward test your pull request locally.

**Comment by briansoe66** [comment]
What is the timeline of this issue? Will the eloquent or foxy debian package be patched with this fix? I'd like to plot sensor_msgs/JointState position[i] etc in ROS2.

**Comment by Michael-Equi** [comment]
@dirk-thomas Is this ready to be merged? It looks like @thomas-moulard made the necessary fix in his most recent commit.

**Comment by dirk-thomas** [comment]
@Michael-Equi I am not maintaining this repository anymore. Maybe the current maintainer can step up?

**Comment by ivanpauno** [comment]
Closing in favor of https://github.com/ros-visualization/rqt_plot/pull/71.

**Pull request closed without merging**